*** glade/glade_project.c	Tue Mar 23 12:53:18 1999
--- glade/glade_project.c	Mon Mar 22 12:57:14 1999
***************
*** 42,50 ****
     "project<project_num>". */
  static gint project_num = 1;
  
! gchar *GladeLanguages[] = { "C", "C++" };
  gint GladeNumLanguages = sizeof (GladeLanguages) / sizeof (GladeLanguages[0]);
  
  #ifndef GTK_HAVE_FEATURES_1_1_0
  typedef void (*GladeProjectSignal1) (GtkObject * object,
  				     GtkObject * arg1,
--- 42,55 ----
     "project<project_num>". */
  static gint project_num = 1;
  
! gchar *GladeLanguages[] = { "C", "C++", "Ada 95" };
  gint GladeNumLanguages = sizeof (GladeLanguages) / sizeof (GladeLanguages[0]);
  
+ gchar *GladeSourceGenerators[] =
+   { NULL,       /* built-in generator for C */
+     "glade--",  /* C++ default code generator */
+     "gate"};    /* Ada 95 default code generator*/
+ 
  #ifndef GTK_HAVE_FEATURES_1_1_0
  typedef void (*GladeProjectSignal1) (GtkObject * object,
  				     GtkObject * arg1,
***************
*** 511,516 ****
--- 516,522 ----
    if (project->language == language)
      return;
    project->language = language;
+   project->source_generator = GladeSourceGenerators[language];
    property_show_lang_specific_page (language);
    glade_project_set_changed (project, TRUE);
  }
*** glade/glade_project.h	Tue Mar 23 12:53:18 1999
--- glade/glade_project.h	Mon Mar 22 12:48:36 1999
***************
*** 39,50 ****
  typedef enum
  {
    GLADE_LANGUAGE_C,
!   GLADE_LANGUAGE_CPP
  } GladeLanguageType;
  
  extern gchar *GladeLanguages[];
  extern gint GladeNumLanguages;
  
  typedef struct _GladeProjectClass  GladeProjectClass;
  
  /* All directories are absolute paths in GladeProject, but some are saved in
--- 39,53 ----
  typedef enum
  {
    GLADE_LANGUAGE_C,
!   GLADE_LANGUAGE_CPP,
!   GLADE_LANGUAGE_ADA95
  } GladeLanguageType;
  
  extern gchar *GladeLanguages[];
  extern gint GladeNumLanguages;
  
+ extern gchar *GladeSourceGenerators[];
+ 
  typedef struct _GladeProjectClass  GladeProjectClass;
  
  /* All directories are absolute paths in GladeProject, but some are saved in
***************
*** 72,77 ****
--- 75,84 ----
       GladeLanguages[]. */
    GladeLanguageType language;
  
+   /* The name of an external source code generator that takes an XML file
+      as an argument. This is particularly useful for languages other than C */
+   gchar *source_generator;
+ 
    /* If the project has changed since the last time it was saved.
       FIXME: this isn't completely supported yet. */
    gboolean changed;
*** glade/source.c	Tue Mar 23 12:53:18 1999
--- glade/source.c	Tue Mar 23 12:53:04 1999
***************
*** 31,36 ****
--- 31,37 ----
  
  #include "gbwidget.h"
  #include "glade_project.h"
+ #include "save.h"
  #include "source.h"
  #include "utils.h"
  #ifdef HAVE_OS2_H
***************
*** 87,99 ****
  {
    gchar *old_locale, *saved_locale;
    GladeStatusCode status;
!      
!   old_locale = setlocale (LC_NUMERIC, NULL);
!   saved_locale = g_strdup (old_locale);
!   setlocale (LC_NUMERIC, "C");
!   status = real_source_write (project);
!   setlocale (LC_NUMERIC, saved_locale);
!   g_free (saved_locale);
    return status;
  }
  
--- 88,116 ----
  {
    gchar *old_locale, *saved_locale;
    GladeStatusCode status;
! 
!   if (project->source_generator == NULL)
!     {
!       old_locale = setlocale (LC_NUMERIC, NULL);
!       saved_locale = g_strdup (old_locale);
!       setlocale (LC_NUMERIC, "C");
!       status = real_source_write (project);
!       setlocale (LC_NUMERIC, saved_locale);
!       g_free (saved_locale);
!     }
!   else
!     {
!       gchar *saved_filename = g_strdup (project->xml_filename);
!       char buf [1024];
! 
!       glade_project_set_xml_filename (project, tmpnam (NULL));
!       save_project_file (project);
!       sprintf (buf, "%s %s", project->source_generator, project->xml_filename);
!       status = system (buf) ? GLADE_STATUS_ERROR : GLADE_STATUS_OK;
!       unlink (project->xml_filename);
!       glade_project_set_xml_filename (project, saved_filename);
!     }
! 
    return status;
  }
  
