*** glade/editor.c.old	Tue Jul 11 16:55:38 2000
--- glade/editor.c	Tue Jul 11 16:53:12 2000
***************
*** 19,25 ****
--- 19,27 ----
  #include <string.h>
  
  #include <gdk/gdkkeysyms.h>
+ #ifndef _WIN32
  #include <gdk/gdkx.h>
+ #endif
  
  #include <gtk/gtk.h>
  
*** glade/save.c.old	Tue Jul 11 16:55:18 2000
--- glade/save.c	Tue Jul 11 16:56:18 2000
***************
*** 115,121 ****
    if (glade_util_file_exists (filename))
      {
        backup_filename = g_strdup_printf ("%s.bak", filename);
! #ifdef __EMX__
        /* for OS/2 rename dosn't work if the dest. file exist ! remove it! */
        status = remove (backup_filename);
  #endif
--- 115,121 ----
    if (glade_util_file_exists (filename))
      {
        backup_filename = g_strdup_printf ("%s.bak", filename);
! #if defined (__EMX__) || defined (_WIN32)
        /* for OS/2 rename dosn't work if the dest. file exist ! remove it! */
        status = remove (backup_filename);
  #endif
*** glade/source.c.old	Tue Jul 11 16:55:24 2000
--- glade/source.c	Tue Jul 11 16:57:16 2000
***************
*** 2834,2839 ****
--- 2834,2842 ----
    if (glade_util_file_exists (filename))
      {
        backup_filename = g_strdup_printf ("%s.bak", filename);
+ #if defined (__EMX__) || defined (_WIN32)
+       remove (backup_filename);
+ #endif
        status = rename (filename, backup_filename);
  
        if (status == -1)
*** glade/utils.c.old	Tue Jul 11 17:38:02 2000
--- glade/utils.c	Tue Jul 11 17:39:44 2000
***************
*** 1277,1283 ****
--- 1277,1287 ----
  	  if (mkdir (directory, 0777) != 0)
  #endif
  	    {
+ #ifndef _WIN32
+ /* This happens under WIN32 when stat is confused by the filename, but this is
+    harmless, since we know that the directory exists after all. */
  	      return glade_error_new_system (_("Couldn't create directory:\n  %s\n"), directory);
+ #endif
  	    }
  	}
        else
*** intl/dcgettext.c.old	Tue Jul 11 17:23:06 2000
--- intl/dcgettext.c	Tue Jul 11 17:25:16 2000
***************
*** 225,230 ****
--- 225,235 ----
  #else
  static int enable_secure;
  # define ENABLE_SECURE (enable_secure == 1)
+ #ifdef _WIN32
+ # define DETERMINE_SECURE \
+   if (enable_secure == 0)						      \
+     enable_secure = -1;
+ #else
  # define DETERMINE_SECURE \
    if (enable_secure == 0)						      \
      {									      \
***************
*** 233,238 ****
--- 238,244 ----
        else								      \
  	enable_secure = -1;						      \
      }
+ #endif
  #endif
  
  /* Look up MSGID in the DOMAINNAME message catalog for the current CATEGORY
