--- giscanner/sourcescanner.py.orig	2012-05-30 12:43:10.616181000 +0200
+++ giscanner/sourcescanner.py	2012-05-30 12:43:18.602109000 +0200
@@ -274,7 +274,7 @@
 
         defines = ['__GI_SCANNER__']
         undefs = []
-        cpp_args = os.environ.get('CC', 'cc').split()
+        cpp_args = os.environ.get('CC', 'gcc').split()
         cpp_args += ['-E', '-C', '-I.', '-']
 
         cpp_args += self._cpp_options
