Index: ./gdk/x11/gdkwindow-x11.c
*** ./gdk/x11/gdkwindow-x11.c.orig	2012-03-05 14:50:28.000000000 -0500
--- ./gdk/x11/gdkwindow-x11.c	2012-03-05 14:50:12.000000000 -0500
***************
*** 4294,4300 ****
          finish_drag (mv_resize);
        break;
  
! #ifdef HAVE_XGENERICEVENTS
      case GenericEvent:
        {
          /* we just assume this is an XI2 event */
--- 4294,4300 ----
          finish_drag (mv_resize);
        break;
  
! #if defined (HAVE_XGENERICEVENTS) && defined (XINPUT_2)
      case GenericEvent:
        {
          /* we just assume this is an XI2 event */
