*** gtk/gtktreeview.c.old	Mon Jan 21 19:02:23 2008
--- gtk/gtktreeview.c	Mon Jan 21 18:46:39 2008
*************** cancel_arrow_animation (GtkTreeView *tre
*** 8528,8533 ****
--- 8528,8536 ----
  
        g_source_remove (tree_view->priv->expand_collapse_timeout);
        tree_view->priv->expand_collapse_timeout = 0;
+ 
+       /* Reset node */
+       tree_view->priv->expanded_collapsed_node = NULL;
      }
  }
  
