--- configure.old	2009-12-22 16:56:30.813761900 +0100
+++ configure	2009-12-22 16:56:47.084666200 +0100
@@ -12759,6 +12759,7 @@ if test "$ac_cs_awk_cr" = "a${ac_cr}b"; 
 else
   ac_cs_awk_cr=$ac_cr
 fi
+ac_cs_awk_cr=$ac_cr
 
 echo 'BEGIN {' >"$tmp/subs1.awk" &&
 _ACEOF
--- src/context.c.old	2009-08-26 12:59:35.000000000 +0200
+++ src/context.c	2009-12-22 17:31:30.306420100 +0100
@@ -1430,7 +1430,7 @@ PyTypeObject PycairoContext_Type = {
   pycairo_methods,                    /* tp_methods */
   0,                                  /* tp_members */
   0,                                  /* tp_getset */
-  &PyBaseObject_Type,                 /* tp_base */
+  0, /* &PyBaseObject_Type, */        /* tp_base */
   0,                                  /* tp_dict */
   0,                                  /* tp_descr_get */
   0,                                  /* tp_descr_set */
--- src/font.c.old	2009-08-26 12:59:35.000000000 +0200
+++ src/font.c	2009-12-22 17:32:24.095564900 +0100
@@ -131,7 +131,7 @@ PyTypeObject PycairoFontFace_Type = {
   0,                                  /* tp_methods */
   0,                                  /* tp_members */
   0,                                  /* tp_getset */
-  &PyBaseObject_Type,                 /* tp_base */
+  0, /* &PyBaseObject_Type, */        /* tp_base */
   0,                                  /* tp_dict */
   0,                                  /* tp_descr_get */
   0,                                  /* tp_descr_set */
@@ -410,7 +410,7 @@ PyTypeObject PycairoScaledFont_Type = {
   scaled_font_methods,                /* tp_methods */
   0,                                  /* tp_members */
   0,                                  /* tp_getset */
-  &PyBaseObject_Type,                 /* tp_base */
+  0, /* &PyBaseObject_Type, */        /* tp_base */
   0,                                  /* tp_dict */
   0,                                  /* tp_descr_get */
   0,                                  /* tp_descr_set */
@@ -595,7 +595,7 @@ PyTypeObject PycairoFontOptions_Type = {
   font_options_methods,               /* tp_methods */
   0,                                  /* tp_members */
   0,                                  /* tp_getset */
-  &PyBaseObject_Type,                 /* tp_base */
+  0, /* &PyBaseObject_Type */         /* tp_base */
   0,                                  /* tp_dict */
   0,                                  /* tp_descr_get */
   0,                                  /* tp_descr_set */
--- src/matrix.c.old	2009-08-26 12:59:35.000000000 +0200
+++ src/matrix.c	2009-12-22 17:33:39.038445300 +0100
@@ -332,7 +332,7 @@ PyTypeObject PycairoMatrix_Type = {
   matrix_methods,                     /* tp_methods */
   0,                                  /* tp_members */
   0,                                  /* tp_getset */
-  &PyBaseObject_Type,                 /* tp_base */
+  0, /* &PyBaseObject_Type */         /* tp_base */
   0,                                  /* tp_dict */
   0,                                  /* tp_descr_get */
   0,                                  /* tp_descr_set */
--- src/path.c.old	2009-08-26 12:59:35.000000000 +0200
+++ src/path.c	2009-12-22 17:32:55.092963600 +0100
@@ -206,7 +206,7 @@ PyTypeObject PycairoPath_Type = {
   0,			        	/* tp_methods */
   0,					/* tp_members */
   0,					/* tp_getset */
-  &PyBaseObject_Type,                   /* tp_base */
+  0, /* &PyBaseObject_Type */           /* tp_base */
   0,					/* tp_dict */
   0,					/* tp_descr_get */
   0,					/* tp_descr_set */
--- src/pattern.c.old	2009-08-26 12:59:35.000000000 +0200
+++ src/pattern.c	2009-12-22 17:33:17.588307800 +0100
@@ -194,7 +194,7 @@ PyTypeObject PycairoPattern_Type = {
   pattern_methods,                    /* tp_methods */
   0,                                  /* tp_members */
   0,                                  /* tp_getset */
-  &PyBaseObject_Type,                 /* tp_base */
+  0, /* &PyBaseObject_Type */         /* tp_base */
   0,                                  /* tp_dict */
   0,                                  /* tp_descr_get */
   0,                                  /* tp_descr_set */
--- src/surface.c.old	2009-08-26 12:59:35.000000000 +0200
+++ src/surface.c	2009-12-22 17:33:59.006573300 +0100
@@ -365,7 +365,7 @@ PyTypeObject PycairoSurface_Type = {
   surface_methods,                    /* tp_methods */
   0,                                  /* tp_members */
   0,                                  /* tp_getset */
-  &PyBaseObject_Type,                 /* tp_base */
+  0, /* &PyBaseObject_Type */         /* tp_base */
   0,                                  /* tp_dict */
   0,                                  /* tp_descr_get */
   0,                                  /* tp_descr_set */
